core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -272,6 +273,8 @@ public void testMetricsForNestedStructFields() throws IOException {
     assertCounts(6, 1L, 0L, metrics);
     assertBounds(6, BinaryType.get(),
         ByteBuffer.wrap("A".getBytes()), ByteBuffer.wrap("A".getBytes()), metrics);
+    assertCounts(7, 1L, 0L, 1L, metrics);
+    assertBounds(7, DoubleType.get(), Double.NaN, Double.NaN, metrics);
```

Comment:
Why are NaN values getting into the lower and upper bounds?

Reply:
This was because I added NaN as the only value in this column during the creation of the record in `buildNestedTestRecord`, and currently this will result in upper and lower bound being both NaN (similar behavior as in [this](https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/TestMetrics.java#L354-L355) test. I added this extra column in order to test NaN handling in metrics modes, and change to this test was a side effect. Do you want me to remove the bound check in this test?

Reply:
So I guess this will continue to happen until we ignore NaN values and keep track of the lower and upper bounds ourselves for Parquet and ORC?

This is fine for now, but I would want this to be correct eventually.